csvcubed/csvcubed/readers/configdeserialiser.py

```diff
@@ -0,0 +1,331 @@
+    if convention_col_counts["measures"] < 1:
+        raise ValueError("The data does not contain any measure columns")
+    if convention_col_counts["units"] < 1:  # needs resolving between Rob and Andrew
+        raise ValueError("The data does not contain any attribute columns")
+
+    for i, column_type in enumerate(column_types):
+        # Build a dict of fields for the identified column types' data-class
+        column_name = data.columns[i]
+        column_data = data.T.iloc[i]
+
+        if column_type == "dimension":
+            column_dict = {
+                "type": column_type,
+                "label": column_name,
+                "code_list": True
+            }
+
+        elif column_type == "observations":
+            column_dict = {
+                "type": column_type,
+                "value": column_name,
+                "datatype": "decimal",
+            }
+
+        elif column_type in ["measures", "units"]:
+            column_dict = {
+                "type": column_type
+            }
+        elif column_type == "attribute":
+            # Note: attribute type columns are currently not supported
+            raise ValueError(
+                "Column type 'Attribute' is not supported when using csv naming convention"
+            )
+
+        else:
+            raise ValueError(f"Column type '{column_type}' is not supported.")
+
+        # As all columns will be new in this scenario get a pd.Series for the column
+
+        try:
+            qb_column = map_column_to_qb_component(
+                column_title=column_name,
+                column=column_dict,
+                data=column_data.astype("category"),
+                json_parent_dir=csv_path.parent.absolute(),
+            )
+            cube_columns.append(qb_column)
+
+        except Exception as err:
+            import traceback
+            traceback.print_exc()
+            log.error(f"{type(err)} exception raised because: {repr(err)}")
+
+    cube_metadata = _metadata_from_dict(
+        {"id": str(csv_path).upper(), "title": csv_path.name.upper().replace("_", " ")}
+    )
+    # TODO - Create the new Cube
+    cube = Cube(metadata=cube_metadata, data=data, columns=cube_columns)
+    # validation_errors = cube.validate()
+    validation_errors = cube.pydantic_validation()
+
+    return cube, validation_errors
+
+
+def get_cube_from_config_json(
+        config_path: Path,
+        csv_path: Path
+) -> Tuple[QbCube, List[ValidationError]]:
+    """
+    Generates a Cube structure from a config.json input.
+    :return: tuple of cube and json schema errors (if any)
+    """
+    data: pd.DataFrame = pd.read_csv(csv_path)
+    config = _load_resource(config_path.resolve())
+    schema = _load_resource(Path(config['$schema']).resolve())
+    try:
+        errors = validate_dict_against_schema(value=config, schema=schema)
+        if errors:
+            return None, errors
+
+    except Exception as err:
+        print(err)
```

Comment:
Let's not suppress errors. Is this something you need to catch here? Can we just leave it to propagate to the top level function and be caught there? 

If we do need this try/catch here can we make the exception caught more specific and ensure that we use the `csvcubed.utils.log.log_exception` method to log the exception properly.